runtime/dancebox/tests/common/mod.rs

```diff
@@ -339,3 +369,21 @@ pub const ALICE: [u8; 32] = [4u8; 32];
 pub const BOB: [u8; 32] = [5u8; 32];
 pub const CHARLIE: [u8; 32] = [6u8; 32];
 pub const DAVE: [u8; 32] = [7u8; 32];
+
+/// Calls `f_loop` with all values in `iter`, and returns the first `Some(value)`.
+/// If only `None` is returned, the value returned by `f_else`  is returned. It
+/// is like a for loop which could have an else block in case no iteration
+/// breaks with a value.
+pub fn for_else<I, F, E, R>(iter: I, mut f_loop: F, f_else: E) -> R
```

Comment:
I believe you can use

```rust
iter.find_map(f_loop).unwrap_or_else(f_else)
```

So you shouldn't need this helper function.

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find_map

Reply:
Oh very neat, didn't knew `find_map` :p